src/Algebra/Definitions.agda

```diff
@@ -118,5 +119,14 @@ RightCancellative _•_ = ∀ {x} y z → (y • x) ≈ (z • x) → y ≈ z
 Cancellative : Op₂ A → Set _
 Cancellative _•_ = (LeftCancellative _•_) × (RightCancellative _•_)
 
+AlmostLeftCancellative : A → Op₂ A → Set _
+AlmostLeftCancellative e _•_ = ∀ {x y z} → ¬ x ≈ e → (x • y) ≈ (x • z) → y ≈ z
+
+AlmostRightCancellative : A → Op₂ A → Set _
+AlmostRightCancellative e _•_ = ∀ {x} y z → ¬ x ≈ e → (y • x) ≈ (z • x) → y ≈ z
```

Comment:
Just looking at this again, I notice the lack of symmetry between implict vs explicit arguments for the left and right cases.  i.e. `{x y z}` vs `{x} y z`. Whether or not `y` and `z` are inferrable depends on the definition of the `_•_`. Because we usually pattern match on the lhs arguments first in our definitions of `_+_`, `_*_` etc it's inferrable where as the rhs arguments are not. However there's no reason that users are pattern matching on the lhs first in their own operator definitions.

It seems to me that we should make these definitions symmetric, i.e. make them both `{x} y z` so they work for both lhs and rhs pattern matching definitions. I know `LeftCancellative` and `RightCancellative` don't do this but I feel like we should probably correct this mistake here.

Reply:
I agree.
How do we organize this change?

I am confused with the state of the commit. 
1. There was a message  "Merge branch 'master' into master.   Verified  v 9303d9d",
2. I do not think that this 9303d9d is already in master. (?)
3. Now, I do not know who needs to push this change for "{x} y z" there.
 
When you suggest something of such sort can you, please, add such details? 
Because when I ask about a detail, I need to wait at least one day for answer.
So each line of the program takes 1-2 days processing.
And a simple project of one week would take 1-2 years to agree.

Is not it better to put at least half of the GCD project at a single PR ?
I would add about 5 not so large modules to Algebra.Properties and 2-3 records to Algebra.Bundles.
Also I would write in what succession to consider this.
May be this way there will be less questions for me of when to push each current thing - ?

Reply:
> How do we organize this change?

I have done it.

> Is not it better to put at least half of the GCD project at a single PR ?
I would add about 5 not so large modules to Algebra.Properties and 2-3 records to Algebra.Bundles.
Also I would write in what succession to consider this.
May be this way there will be less questions for me of when to push each current thing - ?

No it is not. Let's look at some your past large PRs.
- Adding new binary definitions: https://github.com/agda/agda-stdlib/pull/783 took 3 months to merge in
- Adding orderings over binary naturals: https://github.com/agda/agda-stdlib/pull/840 took 3 months to merge in.
- Adding subtraction over binary naturals: https://github.com/agda/agda-stdlib/pull/888 - https://github.com/agda/agda-stdlib/pull/1127 took **7** months to merge in.

In contrast this PR has taken 10 days to merge in. Hopefully think the numbers speak for themselves! Please continue to make small PRs.

Reply:
> But I do not know, I recall, may be I can do this myself by making a branch ...

Yes you could do this with branches, and open many small independent PRs at the same time. That would indeed be quicker.